### PR TITLE
Missing application_properties in BatchMessage fixes

### PR DESCRIPTION
**Pull request by yunhaoling**
https://github.com/Azure/azure-event-hubs-python/issues/97

### CODE REVIEW

File: uamqp/message.py [hunk 1]
```diff
@@ -458,25 +458,32 @@ def _multi_message_generator(self):
...
                 for data in self._body_gen:
-                    message_segment = []
-                    if isinstance(data, six.text_type):
-                        data = data.encode(self._encoding)
-                    batch_data = c_uamqp.create_data(data)
-                    c_uamqp.enocde_batch_value(batch_data, message_segment)
-                    combined = b"".join(message_segment)
-                    body_size += len(combined)
+                    message_bytes = None
+                    if not isinstance(data, Message):  # raw data
```
Comment (annatisch) [inline uamqp/message.py]: After giving it some thought - I think this type check could be easier to adapt to the EH SDK if we do it like this:
```python
try:
    if not data.application_properties:  # Message-like object
        data.application_properties = self.application_properties
    message_bytes = data.encode_message()
except AttributeError:  # raw data
    wrap_message = Message(body=data, application_properties=self.application_properties)
    message_bytes = wrap_message.encode_message()
```
This means we could consume any "message-like" object, so long as it has an "application_properties" attribute and an "encode_message" function. This should make it easier to adapt the EH SDK, as an `Event` object already has the `application_properties` attribute, we just need to give it an `encode_message` function and we should be able to use it directly.

File: uamqp/message.py [hunk 2]
```diff
@@ -501,19 +508,20 @@ def gather(self):
...
-            message_segment = []
-            if isinstance(data, six.text_type):
-                data = data.encode(self._encoding)
-            batch_data = c_uamqp.create_data(data)
-            c_uamqp.enocde_batch_value(batch_data, message_segment)
-            combined = b"".join(message_segment)
-            body_size += len(combined)
-
+            message_bytes = None
+            if not isinstance(data, Message):  # raw data
```
Comment (annatisch) [inline uamqp/message.py]: Same type-check comment as above.

File: uamqp/message.py (hunk 1 above)
Comment (yunhaoling) [reply uamqp/message.py]: Great idea! Get what you mean and it reminds me of duck typing.
This is undoubtedly a better implementation. I think what we should do after taking this implementation is to provide more detailed documentations about batching in EventData.

**Comment by annatisch** [comment]
Hi @yunhaoling - I updated pylint as it was throwing errors due a version conflict, however this has resulted in some new formatting errors. Could you also fix these? There's not too many:
https://travis-ci.org/Azure/azure-uamqp-python/jobs/511066675

**Comment by yunhaoling** [comment]
@annatisch Sure, I'll investigate into that.

**Comment by annatisch** [comment]
Hi @yunhaoling - I would revert your last two commits and simply remove Python 3.4 from the travis config:
https://github.com/Azure/azure-uamqp-python/blob/master/.travis.yml#L14

We don't plan to support Python 3.4 from the next release anyway :)

**Comment by yunhaoling** [comment]
Hey @annatisch 
I've done the rollback and removed python 3.4 from travis config.

**Pull request merged**